WebAPI/Controllers/UserController.cs

```diff
@@ -9,58 +13,98 @@ namespace WebAPI.Controllers
 {
     [Produces("application/json")]
     [Route("api/[controller]")]
-    public class UserController : Controller
+    public class UserController : BaseController
     {
-        private ILogger<UserController> _logger;
-        private readonly IUserService _userService;
+        private readonly ILogger<UserController> _logger;
+        private readonly IEmailService _emailService;
+        private readonly UserManager<User> _userManager;
+        private readonly SignInManager<User> _signInManager;
 
-        public UserController(ILogger<UserController> logger, IUserService userService)
+        public UserController(ILogger<UserController> logger, UserManager<User> userManager, SignInManager<User> signInManager,
+            IEmailService emailService)
         {
             _logger = logger;
-            _userService = userService;
+            _userManager = userManager;
+            _signInManager = signInManager;
+            _emailService = emailService;
         }
 
-        // GET: api/User
-        [HttpGet]
-        public IEnumerable<string> Get()
+        [HttpPost("register")]
+        public async Task<IActionResult> Register([FromBody] RegisterViewModel body)
```

Comment:
We shoul make clear, do we use MVC o MVVM?
Probably I will need some explanation about the whole architecture, to be able to review the C# code also in the future :-)